### ISSUE TITLE
Audit places which assume float element type

### ISSUE THREAD

**Issue by asuhan** [pytorch/xla#203]
A few operations (threshold, batch norm) silently cast to float and they should be agnostic to float vs double. For example: https://github.com/pytorch/xla/blob/master/torch_xla/csrc/ops/threshold.h.

**Comment by asuhan** [comment]
We have plenty of tests which verify this isn't a problem anymore.
